frontend/server/src/Controllers/Course.php

```diff
@@ -134,6 +138,44 @@ private static function validateCreateAssignment(
             'alias',
             /*$isRequired=*/true
         );
+
+        $addedProblems = [];
+        if (!empty($r['problems'])) {
+          /** @var list<array{alias: string, points: int|null}> */
+            $problemsData = json_decode(
+                strval(
+                    $r['problems']
+                ), /*$assoc=*/
+                true
+            );
+            foreach ($problemsData as $problemData) {
+                if (!isset($problemData['alias'])) {
+                    throw new \OmegaUp\Exceptions\InvalidParameterException(
+                        'parameterEmpty',
+                        'problems'
+                    );
+                }
+
+                if (
+                    is_null(
+                        \OmegaUp\DAO\Problems::getByAlias(
+                            $problemData['alias']
+                        )
+                    )
+                ) {
+                    throw new \OmegaUp\Exceptions\NotFoundException(
+                        'problemNotFound'
+                    );
+                }
+
+                if (!isset($problemData['points'])) {
+                    $problemData['points'] = 100; // Set 100 as default
+                }
```

Comment:
```suggestion
                $problemData['points'] = $problemData['points']) ?? 100;
```

eso es un poco más idiomático, así que ya no sería necesario el comentario.

Reply:
Listo, aunque tuve que regresarlo como estaba para poder hacer la validación de que `['points']` es un número válido.